src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsLocationMapping.java

```diff
@@ -0,0 +1,151 @@
+                                                            throws InvalidMappingException {
+        List<HearingLocation> locations = getOverrideLocations(caseData);
+
+        if (isNotEmpty(locations)) {
+            return locations;
+        }
+
+        locations = getPaperCaseLocations(caseData, referenceDataServiceHolder);
+
+        if (isNotEmpty(locations)) {
+            return locations;
+        }
+
+        locations = getAdjournedLocations(caseData, referenceDataServiceHolder);
+
+        if (isNotEmpty(locations)) {
+            return locations;
+        }
+
+        return getMultipleLocations(caseData, referenceDataServiceHolder);
+    }
+
+    private static List<HearingLocation> getOverrideLocations(SscsCaseData caseData) {
+        OverrideFields overrideFields = OverridesMapping.getOverrideFields(caseData);
+
+        if (isNotEmpty(overrideFields.getHearingVenueEpimsIds())) {
+            return overrideFields.getHearingVenueEpimsIds().stream()
+                .map(CcdValue::getValue)
+                .map(CcdValue::getValue)
+                .map(epimsId -> HearingLocation.builder()
+                    .locationId(epimsId)
+                    .locationType(COURT)
+                    .build())
+                .collect(Collectors.toList());
+        }
+
+        return Collections.emptyList();
+    }
+
+    private static List<HearingLocation> getPaperCaseLocations(SscsCaseData caseData,
+                                                               ReferenceDataServiceHolder referenceDataServiceHolder) {
+        if (HearingsChannelMapping.isPaperCase(caseData)) {
+            List<VenueDetails> venueDetailsList = referenceDataServiceHolder
+                .getVenueService()
+                .getActiveRegionalEpimsIdsForRpc(caseData.getRegionalProcessingCenter().getEpimsId());
+
+            log.info("Found {} venues under RPC {} for paper case {}", venueDetailsList.size(),
+                     caseData.getRegionalProcessingCenter().getName(), caseData.getCcdCaseId());
+
+            return venueDetailsList.stream()
+                .map(VenueDetails::getEpimsId)
+                .map(id -> HearingLocation.builder()
+                    .locationId(id)
+                    .locationType(COURT)
+                    .build())
+                .collect(Collectors.toList());
+        }
+
+        return Collections.emptyList();
+    }
+
+    private static List<HearingLocation> getAdjournedLocations(SscsCaseData caseData,
+                                                               ReferenceDataServiceHolder referenceDataServiceHolder)
+        throws InvalidMappingException {
+        String nextHearingVenueName = caseData.getAdjournCaseNextHearingVenue();
+
+        //TODO: SSCS-10951: remove adjournment flag
+        if (referenceDataServiceHolder.isAdjournmentFlagEnabled() && isNotEmpty(nextHearingVenueName)) {
```

Comment:
This is not enough to show that the hearing is adjourned. You should use isYes(adjournmentInProgress)